bigquery/src/test/java/org/apache/iceberg/gcp/bigquery/TestBigQueryTableOperations.java

```diff
@@ -0,0 +1,291 @@
+  private static final String GCP_PROJECT = "my-project";
+  private static final String GCP_REGION = "us";
+  private static final String NS = "db";
+  private static final String TABLE = "tbl";
+  private static final TableIdentifier IDENTIFIER = TableIdentifier.of(NS, TABLE);
+
+  private static final TableReference TABLE_REFERENCE =
+      new TableReference().setProjectId(GCP_PROJECT).setDatasetId(NS).setTableId(TABLE);
+
+  private static final Schema SCHEMA =
+      new Schema(
+          required(1, "id", Types.IntegerType.get(), "unique ID"),
+          required(2, "data", Types.StringType.get()));
+
+  private final BigQueryMetastoreClient client = mock(BigQueryMetastoreClient.class);
+
+  private BigQueryMetastoreCatalog catalog;
+  private BigQueryTableOperations tableOps;
+
+  @BeforeEach
+  public void before() {
+    this.catalog = new BigQueryMetastoreCatalog();
+    this.catalog.setConf(new Configuration());
+    String warehouseLocation = tempFolder.toPath().resolve("hive-warehouse").toString();
+
+    catalog.initialize(
+        "CATALOG_ID",
+        ImmutableMap.of(
+            PROJECT_ID,
+            GCP_PROJECT,
+            CatalogProperties.WAREHOUSE_LOCATION,
+            warehouseLocation,
+            CatalogProperties.FILE_IO_IMPL,
+            "org.apache.iceberg.hadoop.HadoopFileIO"),
+        GCP_PROJECT,
+        GCP_REGION,
+        client);
+    this.tableOps = (BigQueryTableOperations) catalog.newTableOps(IDENTIFIER);
+  }
+
+  @Test
+  public void fetchLatestMetadataFromBigQuery() throws Exception {
+    Table createdTable = createTestTable();
+    reset(client);
+    when(client.load(TABLE_REFERENCE)).thenReturn(createdTable);
+
+    tableOps.refresh();
+    assertThat(
+            createdTable
+                .getExternalCatalogTableOptions()
+                .getParameters()
+                .getOrDefault(METADATA_LOCATION_PROP, ""))
+        .isEqualTo(tableOps.currentMetadataLocation());
+
+    reset(client);
+    when(client.load(TABLE_REFERENCE))
+        .thenThrow(new NoSuchTableException("error message getTable"));
+    // Refresh fails when table is not found but metadata already presents.
+    assertThatThrownBy(() -> tableOps.refresh())
+        .isInstanceOf(NoSuchTableException.class)
+        .hasMessageContaining("error message getTable");
+  }
+
+  @Test
+  public void loadNonIcebergTableFails() {
+    when(client.load(TABLE_REFERENCE)).thenReturn(new Table().setTableReference(TABLE_REFERENCE));
+
+    assertThatThrownBy(() -> tableOps.refresh())
+        .isInstanceOf(ValidationException.class)
+        .hasMessageContaining("metadata location not found");
+  }
+
+  @Test
+  public void loadNoOpWhenMetadataAndTableNotFound() {
+    when(client.load(TABLE_REFERENCE))
+        .thenThrow(new NoSuchTableException("error message getTable"));
+    // Table not found won't cause errors when the metadata is null.
+    assertThat(tableOps.currentMetadataLocation()).isNull();
```

Comment:
previously you had `tableOps.refresh()` after this line and I think you wanted to keep it as otherwise the NSTE won't be thrown. That's also why I mentioned that it would be better to wrap this in a `assertThatNoException().isThrownBy(..)` to make it explicit about what's expected in this test

Reply:
Yes I added. While applying changes Looks like I deleted but i did not added :)